pkg/services/ngalert/schedule/schedule.go

```diff
@@ -454,10 +452,16 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key models.AlertRul
 				processedStates := sch.stateManager.ProcessEvalResults(alertRule, results)
 				sch.saveAlertStates(processedStates)
 				alerts := FromAlertStateToPostableAlerts(sch.log, processedStates, sch.stateManager, sch.appURL)
-				sch.log.Debug("sending alerts to notifier", "count", len(alerts.PostableAlerts), "alerts", alerts.PostableAlerts)
-				err = sch.notifier.PutAlerts(alerts)
-				if err != nil {
-					sch.log.Error("failed to put alerts in the notifier", "count", len(alerts.PostableAlerts), "err", err)
+
+				sch.log.Debug("sending alerts to notifier", "count", len(alerts.PostableAlerts), "alerts", alerts.PostableAlerts, "org", alertRule.OrgID)
+				n, err := sch.multiOrgNotifier.AlertmanagerFor(alertRule.OrgID)
+				if err == nil {
+					err = n.PutAlerts(alerts)
```

Comment:
I noticed that we are assigning to an err variable that is scoped outside of this if statement, perhaps we could do `if err := n.PutAlerts(alerts); err != nil` to prevent err from leaking out of this if statement?

Reply:
Good catch!